### DISCUSSION TITLE
Ingesting and Parsing syslogs via wazuh

### DISCUSSION THREAD

**Discussion by pkgwillie** [Security-Onion-Solutions/securityonion#3119] [Unsupported Versions]
I am trying to ship Cisco ASA logs via syslog (udp/514) to securityonion standalone instance. I want to use in built wazuh decoders and rules to automatically extract fields from message and also generate alerts based on logs.

This is what i have done so far: in 
 /nsm/wazuh/etc/ossec.conf
add lines below 

`<remote>
  <connection>syslog</connection>
  <port>514</port>
  <protocol>udp</protocol>
  <allowed-ips>x.x.x.x</allowed-ips>
  <local_ip>x.x.x.x</local_ip>
</remote>`

Tcpdump shows that logs are reaching the SO box. When I check the Discover tab in Kibana, the logs are being processed by the default elasticsearch ingest pipeline for syslog.

Can anyone assist on how to get syslogs from cisco asa to be processed by wazuh decoders and rulesets ?

**Comment by weslambert** [comment]
In order to have Wazuh analyze those logs, I believe you would need to configure Wazuh itself to listen on an appropriate port, and the Docker container would need to be allowed to bind to that port.  In short, it sounds like you would need to modify:

Modify `/nsm/wazuh/etc/ossec.conf` with the new port assignment for syslog.

`cp /opt/so/saltstack/default/salt/wazuh/init.sls /opt/so/saltstack/local/salt/wazuh/init.sls` and modify to include the new port, restarting Wazuh after.

**Reply by pkgwillie** [reply]
Thanks Wes,

This was very helpful.
I have done a new port assignment for syslog (port tcp/udp/1600). 
I also included new port and restarted service. Netstat shows udp/tcp ports listening on port 1600
Tcpdump also shows that logs are arriving from Cisco Asa (10.120.16.2) to the SO box.

19:39:18.317274 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.info, length: 145
19:39:18.317338 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.debug, length: 94
19:39:18.317352 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.info, length: 204
19:39:18.334094 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.debug, length: 134
19:39:18.383410 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.info, length: 174
19:39:18.442853 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.debug, length: 132
19:39:18.500796 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.info, length: 183
19:39:18.500853 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.info, length: 181
19:39:18.500868 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.debug, length: 115
19:39:18.511718 IP 10.120.16.2.syslog > SIEM.issd: SYSLOG local4.debug, length: 127
19:39:18.550110 IP gateway.syslog > SIEM.issd: SYSLOG local4.warning, length: 163
19:39:18.622897 IP gateway.syslog > SIEM.issd: SYSLOG local4.warning, length: 202
19:39:18.748510 IP gateway.syslog > SIEM.issd: SYSLOG local4.warning, length: 165
19:39:18.748564 IP gateway.syslog > SIEM.issd: SYSLOG local4.warning, length: 165
19:39:18.749710 IP gateway.syslog > SIEM.issd: SYSLOG local4.warning, length: 165
19:39:18.774215 IP gateway.syslog > SIEM.issd: SYSLOG local4.warning, length: 159

When I check Kibana however, I don't see any logs from the cisco asa ip. Is there any step am missing ?

**Reply by pkgwillie** [reply]
@weslambert  is there anything am missing?|

**Comment by pkgwillie** [comment]
Below are are the changes made.

ossec.conf
------------------
` <remote>
    <connection>syslog</connection>
    <port>1600</port>
    <protocol>udp</protocol>
    <allowed-ips>10.120.16.2</allowed-ips>
  </remote>`


init.sls
--------------
so-wazuh:
  docker_container.running:
    - image: {{ MANAGER }}:5000/{{ IMAGEREPO }}/so-wazuh:{{ VERSION }}
    - hostname: {{HOSTNAME}}-wazuh-manager
    - name: so-wazuh
    - detach: True
    - port_bindings:
      - 0.0.0.0:1514:1514/udp
      - 0.0.0.0:1514:1514/tcp
      - 0.0.0.0:1515:1515/tcp
      - 0.0.0.0:1600:1600/udp
      - 0.0.0.0:1600:1600/tcp
      - 0.0.0.0:55000:55000
    - binds:
      - /nsm/wazuh:/var/ossec/data:rw



Netstat
--------------
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:4505            0.0.0.0:*               LISTEN      2309/python3
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      2196/master
tcp        0      0 0.0.0.0:4506            0.0.0.0:*               LISTEN      2316/python3
tcp        0      0 127.0.0.1:1234          0.0.0.0:*               LISTEN      19109/stenographer
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1596/sshd
tcp6       0      0 :::8090                 :::*                    LISTEN      10418/docker-proxy
tcp6       0      0 :::443                  :::*                    LISTEN      10432/docker-proxy
tcp6       0      0 :::9500                 :::*                    LISTEN      26293/docker-proxy
tcp6       0      0 :::9822                 :::*                    LISTEN      11008/docker-proxy
tcp6       0      0 :::1600                 :::*                    LISTEN      31608/docker-proxy
tcp6       0      0 :::3200                 :::*                    LISTEN      27452/docker-proxy
tcp6       0      0 :::9696                 :::*                    LISTEN      17972/docker-proxy
tcp6       0      0 :::9600                 :::*                    LISTEN      17418/docker-proxy
tcp6       0      0 :::5601                 :::*                    LISTEN      18501/docker-proxy
tcp6       0      0 :::514                  :::*                    LISTEN      24769/docker-proxy
tcp6       0      0 :::57314                :::*                    LISTEN      23954/docker-proxy
tcp6       0      0 :::6050                 :::*                    LISTEN      17494/docker-proxy
tcp6       0      0 :::6051                 :::*                    LISTEN      17474/docker-proxy
tcp6       0      0 :::6052                 :::*                    LISTEN      17456/docker-proxy
tcp6       0      0 :::6053                 :::*                    LISTEN      17436/docker-proxy
tcp6       0      0 :::3142                 :::*                    LISTEN      5044/docker-proxy
tcp6       0      0 :::9000                 :::*                    LISTEN      27188/docker-proxy
tcp6       0      0 :::5000                 :::*                    LISTEN      5008/docker-proxy
tcp6       0      0 :::9001                 :::*                    LISTEN      26798/docker-proxy
tcp6       0      0 :::1514                 :::*                    LISTEN      31650/docker-proxy
tcp6       0      0 :::3306                 :::*                    LISTEN      13766/docker-proxy
tcp6       0      0 :::1515                 :::*                    LISTEN      31636/docker-proxy
tcp6       0      0 :::6379                 :::*                    LISTEN      17991/docker-proxy
tcp6       0      0 :::6380                 :::*                    LISTEN      23149/docker-proxy
tcp6       0      0 :::5644                 :::*                    LISTEN      17511/docker-proxy
tcp6       0      0 :::6381                 :::*                    LISTEN      23771/docker-proxy
tcp6       0      0 :::8080                 :::*                    LISTEN      25589/docker-proxy
tcp6       0      0 :::9200                 :::*                    LISTEN      14517/docker-proxy
tcp6       0      0 :::80                   :::*                    LISTEN      10447/docker-proxy
tcp6       0      0 :::47761                :::*                    LISTEN      22990/zeek
tcp6       0      0 :::4433                 :::*                    LISTEN      11135/docker-proxy
tcp6       0      0 :::47762                :::*                    LISTEN      23094/zeek
tcp6       0      0 :::4434                 :::*                    LISTEN      11121/docker-proxy
tcp6       0      0 :::47763                :::*                    LISTEN      23234/zeek
tcp6       0      0 :::47764                :::*                    LISTEN      23343/zeek
tcp6       0      0 :::5044                 :::*                    LISTEN      17531/docker-proxy
tcp6       0      0 :::9300                 :::*                    LISTEN      14505/docker-proxy
tcp6       0      0 :::47765                :::*                    LISTEN      23339/zeek
tcp6       0      0 :::47766                :::*                    LISTEN      23340/zeek
tcp6       0      0 :::8086                 :::*                    LISTEN      10711/docker-proxy
tcp6       0      0 :::22                   :::*                    LISTEN      1596/sshd
tcp6       0      0 :::55000                :::*                    LISTEN      31593/docker-proxy
tcp6       0      0 :::9400                 :::*                    LISTEN      26307/docker-proxy
tcp6       0      0 :::7000                 :::*                    LISTEN      25896/docker-proxy
tcp6       0      0 :::3000                 :::*                    LISTEN      10868/docker-proxy
udp        0      0 127.0.0.1:323           0.0.0.0:*                           926/chronyd
udp6       0      0 ::1:323                 :::*                                926/chronyd
udp6       0      0 :::514                  :::*                                24784/docker-proxy
udp6       0      0 :::1514                 :::*                                31663/docker-proxy
udp6       0      0 :::1600                 :::*                                31622/docker-proxy

**Reply by thehien2205** [reply]
You should add firewall rule for port 1600

**Reply by pkgwillie** [reply]
@thehien2205 the firewall rule for port 1600 has already been added.
